DQM/BeamMonitor/plugins/BeamSpotDipServer.cc

```diff
@@ -0,0 +1,770 @@
+}
+
+/*****************************************************************************/ 
+string BeamSpotDipServer::tkStatus()
+{
+  string outstr;
+
+  if(readFromNFS)
+  { // get from file on /nfs
+    ifstream logfile(sourceFile1);
+
+    if (!logfile.good() || getFileSize(sourceFile1) == 0) {
+      // file does not exist or has zero size
+      outstr = "No CMS Tracker status available. No DAQ/DQM.";
+    } else {
+      int nthLnInRcd = 0;
+      string record;
+
+      try {
+        string record;
+
+        while (getline(logfile,record)) {
+          nthLnInRcd ++;
+          vector<string> tmp = parse(record, " ");
+
+          switch(nthLnInRcd) {
+            case 7  :
+              if (tmp[1].find("Yes") == string::npos)
+                outstr = "CMS Tracker OFF.";
+              else
+                outstr = "CMS not taking data or no beam.";
+              break;
+            case 8  : runnum = stoi(tmp[1]); break;
+            default : break;
+          }
+        }
+      } catch (exception & e) {
+        edm::LogWarning("BeamSpotDipServer")
+          << "exception (tkStatus): " << e.what();
+      }
+    }
+
+    logfile.close();
+  }
+  else
+  { 
+    // get from DCS
+    if(wholeTrackerOn) outstr = "CMS not taking data or no beam.";
+                  else outstr = "CMS Tracker OFF.";
+  }
+
+  return outstr;
+}
+
+/*****************************************************************************/ 
+void BeamSpotDipServer::problem()
+{
+  if(verbose)
+    edm::LogInfo("BeamSpotDipServer") << "no update | alive = " << alive;
+
+  lsCount++;
+
+  if ((lsCount % timeoutLS[0] == 0)
+   && (lsCount % timeoutLS[1] != 0)) // first time out
+  {
+    if (!alive.test(1)) alive.flip(1);
+    if (!alive.test(2))
+    {
+      if (!alive.test(7)) fakeRcd();
+                     else trueRcd();
+
+      stringstream warnMsg;
+      warnMsg << "No new data for " << lsCount << " LS";
+      publishRcd("Uncertain",warnMsg.str(),false,false);
+    } else {
+      fakeRcd();
+
+      stringstream warnMsg;
+      warnMsg << "No new data for " << lsCount << " LS: "
+               << tkStatus();
+      publishRcd("Bad",warnMsg.str(),false,false);
+    }
+  }
+  else if (lsCount % timeoutLS[1] == 0) // second time out
+  {
+    if (!alive.test(2)) alive.flip(2);
+    fakeRcd();
+
+    stringstream warnMsg;
+    warnMsg << "No new data for " << lsCount << " LS: " << tkStatus();
+    publishRcd("Bad",warnMsg.str(),false,false);
+  }
+}
+
+/*****************************************************************************/ 
+bool BeamSpotDipServer::readRcd(const BeamSpotOnlineObjects & bs)
+// read from database
+{
+  runnum = bs.GetLastAnalyzedRun();
+
+  // BeamSpotOnlineObject only provides creation time
+
+  startTime      = bs.GetCreationTime(); // provides cond::Time_t -> string ?
+  startTimeStamp = bs.GetCreationTime(); // provides cond::Time_t -> time_t ?
+    endTime      = bs.GetCreationTime();
+    endTimeStamp = bs.GetCreationTime();
```

Comment:
Please use the methods from the [BeamSpotOnlineObjects class](https://github.com/cms-sw/cmssw/blob/master/CondFormats/BeamSpotObjects/interface/BeamSpotOnlineObjects.h) recently instroduced in #35405:
 - GetStartTime()
 - GetEndTime()
 - GetStartTimeStamp()
 - GetEndTimeStamp()